src/portable/synopsys/dwc2/dcd_dwc2.c

```diff
@@ -1207,10 +1207,6 @@ bool dcd_check_test_mode_support(test_mode_t test_selector) {
 }
 
 void dcd_enter_test_mode(uint8_t rhport, test_mode_t test_selector) {
-  // Disable test mode if not supported as a fall back
-  if (!dcd_check_test_mode_support(test_selector)) {
-    test_selector = 0;
-  }
```

Comment:
This is not dead code! This is a check if the used test selector is actually supported and to fall back to do nothing in that case.

The app can still call `dcd_enter_test_mode()` directly and not all chips which are compatible with DWC2 actually support test modes.

Reply:
dcd layer are private functions can't be called by user code. If there is a need of force enable should implement a usbd_ function